src/core/Song.cpp

```diff
@@ -1180,54 +1208,42 @@ bool Song::saveProjectFile( const QString & filename )
 
 	m_savingProject = false;
 
-	return dataFile.writeFile( filename );
+	return dataFile.writeFile(filename, withResources);
 }
 
 
 
 // Save the current song
 bool Song::guiSaveProject()
 {
-	DataFile dataFile( DataFile::SongProject );
-	QString fileNameWithExtension = dataFile.nameWithExtension( m_fileName );
-	setProjectFileName(fileNameWithExtension);
-
-	bool const saveResult = saveProjectFile( m_fileName );
-
-	if( saveResult )
-	{
-		setModified(false);
-	}
-
-	return saveResult;
+	return guiSaveProjectAs(m_fileName);
```

Comment:
Got it, the change was very linked to the feature though, not exactly a refactoring. I had to move things around a bit to make it easier to add the `withResources` parameter to the project save. Maybe a separate commit would be better but I think I needed that change to make the feature work so it might have gone in a single commit.

It ended up making the two methods cleaner I think: Before, `saveAs` called `save` changing the project name first to the new project name. Now `save` calls `saveAs` with the same project name, which I think makes more sense.

Reply:
There are just many questions I have. Before, the two methods were much different, so putting them together probably changed some logic?

Reply:
It changed slightly, I'll try to go through the changes and write down why they were made and post it here or on Discord! I'll have to read everything to remember.

Was a bit stuck yesterday working on the `hasLocalPlugin` method. I need to use recursion to go through all XML nodes but I think using a lambda function with recursion is more complicated than I thought, so I'll probably rewrite the method itself as recursive :grimacing:

Reply:
If you write it here or in Discord, future coders who find bugs in refactoring of this function will not find the description. Such justifications should IMO always go into the commit message.

Reply:
I'll write it in the PR body. Since we squash and merge those commit messages are usually lost anyways, except for the PR log. I tend to be detailed on my commit messages, but I didn't mention those changes specifically probably because I didn't think they were very critical. IIRR the changes were more organizational then functional, just to make calling the modified `DataFile::writeFile` with the new parameter possible while also keeping the bundle exclusive to the `Save as` action.

But I'll still read the code again and write a justification for the changes so it's easily referenced in the PR.